plugins/woocommerce/tests/php/includes/settings/class-wc-settings-tax-test.php

```diff
@@ -172,4 +172,171 @@ public function test_save_tax_classes() {
+		update_option( 'woocommerce_calc_taxes', 'yes' );
+		update_option( 'woocommerce_default_country', 'US:CA' );
+
+		// Mock the screen to simulate being on WooCommerce settings page.
+		set_current_screen( 'woocommerce_page_wc-settings' );
+
+		// Mock WC_Tax methods to return empty base rates.
+		StaticMockerHack::add_method_mocks(
+			array(
+				'WC_Tax' => array(
+					'get_tax_classes'    => function () {
+						return array();
+					},
+					'get_base_tax_rates' => function () {
+						return array();
+					},
+				),
+			)
+		);
+
+		$sut = new WC_Settings_Tax();
+
+		// Capture output.
+		ob_start();
+		$sut->tax_configuration_validation_notice();
+		$output = ob_get_clean();
+
+		// Assert notice is displayed.
+		$this->assertStringContainsString( 'Tax configuration incomplete', $output );
+		$this->assertStringContainsString( 'configure tax rates', $output );
+		$this->assertStringContainsString( 'notice-warning', $output );
+	}
+
+	/**
+	 * @testDox 'tax_configuration_validation_notice' does not show notice when base rate exists.
+	 */
+	public function test_tax_configuration_validation_notice_does_not_show_when_base_rate_exists() {
+		// Set up prices include tax option.
+		update_option( 'woocommerce_prices_include_tax', 'yes' );
+		update_option( 'woocommerce_calc_taxes', 'yes' );
+		update_option( 'woocommerce_default_country', 'US:CA' );
+
+		// Mock the screen.
+		set_current_screen( 'woocommerce_page_wc-settings' );
+
+		// Mock WC_Tax methods to return base rates.
+		StaticMockerHack::add_method_mocks(
+			array(
+				'WC_Tax' => array(
+					'get_tax_classes'    => function () {
+						return array();
+					},
+					'get_base_tax_rates' => function () {
+						return array(
+							array(
+								'rate' => '10.0000',
+							),
+						);
+					},
+				),
+			)
+		);
+
+		$sut = new WC_Settings_Tax();
+
+		// Capture output.
+		ob_start();
+		$sut->tax_configuration_validation_notice();
+		$output = ob_get_clean();
+
+		// Assert notice is NOT displayed.
+		$this->assertEmpty( $output );
+	}
+
+	/**
+	 * @testDox 'tax_configuration_validation_notice' does not show notice when prices are not tax-inclusive.
+	 */
+	public function test_tax_configuration_validation_notice_does_not_show_when_prices_not_inclusive() {
+		// Set up prices exclude tax option.
+		update_option( 'woocommerce_prices_include_tax', 'no' );
+		update_option( 'woocommerce_calc_taxes', 'yes' );
+
+		// Mock the screen.
+		set_current_screen( 'woocommerce_page_wc-settings' );
+
+		$sut = new WC_Settings_Tax();
+
+		// Capture output.
+		ob_start();
+		$sut->tax_configuration_validation_notice();
+		$output = ob_get_clean();
+
+		// Assert notice is NOT displayed.
+		$this->assertEmpty( $output );
+	}
+
+	/**
+	 * @testDox 'tax_configuration_validation_notice' does not show notice on non-WooCommerce pages.
+	 */
+	public function test_tax_configuration_validation_notice_does_not_show_on_non_woocommerce_pages() {
```

Comment:
As this is public we should check that taxes are enabled and return if otherwise:

```
  if ( ! wc_tax_enabled() ) {
        return;
    }
```